app/models/article.rb

```diff
@@ -338,11 +338,14 @@ def search_id
   end
 
   def processed_description
-    text_portion = body_text.present? ? body_text[0..100].tr("\n", " ").strip.to_s : ""
-    text_portion = "#{text_portion.strip}..." if body_text.size > 100
-    return "A post by #{user.name}" if text_portion.blank?
-
-    text_portion.strip
+    if body_text.present?
+      body_text
+        .truncate(104, separator: " ")
```

Comment:
This should be 100, right? I think `.truncate` adds the ellipsis after shortening the string.

Reply:
you would think it should be 100 - I sure did! But the original behavior was "grab 101 characters from 0 to 100 inclusive, strip, and then add ellipsis to it, total of up to 104 characters".

I can set it to 100 (the article model spec only checks "ends with ellipsis dots") but there's [an api spec](https://github.com/forem/forem/blob/main/spec/requests/api/v0/articles_spec.rb#L859) that asserts on the full string being slightly longer than the 100 character version truncate gave. It's very possible the original implementation did something different than what the author expected it to.

Reply:
Ah... interesting! Thanks for pointing that out. Matching the original behavior as best as you can seems to make sense. I guess I'm more in favor of changing it to 100 because it's easier to read, whereas seeing 104 makes me wonder why it's 104.

That said I don't care too much here either. 😅